File: samples/client/petstore/java/retrofit2rx2/src/test/java/io/swagger/client/api/FakeApiTest.java
package io.swagger.client.api;

import io.swagger.client.ApiClient;
import java.math.BigDecimal;
import io.swagger.client.model.Client;
import org.joda.time.DateTime;
import org.joda.time.LocalDate;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * API tests for FakeApi
 */
public class FakeApiTest {

    private FakeApi api;

    @Before
    public void setup() {
        api = new ApiClient().createService(FakeApi.class);
    }

    
    /**
     * To test \&quot;client\&quot; model
     *
     * To test \&quot;client\&quot; model
     */
    @Test
    public void testClientModelTest() {
        Client body = null;
        // Client response = api.testClientModel(body);

        // TODO: test validations
    }
    
    /**
     * Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 
     *
     * Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 
     */
    @Test
    public void testEndpointParametersTest() {
        BigDecimal number = null;
        Double _double = null;
        String patternWithoutDelimiter = null;
        byte[] _byte = null;
        Integer integer = null;
        Integer int32 = null;
        Long int64 = null;
        Float _float = null;
        String string = null;
        byte[] binary = null;
        LocalDate date = null;
        DateTime dateTime = null;
        String password = null;
        String paramCallback = null;
        // Void response = api.testEndpointParameters(number, _double, patternWithoutDelimiter, _byte, integer, int32, int64, _float, string, binary, date, dateTime, password, paramCallback);

        // TODO: test validations
    }
    
    /**
     * To test enum parameters
     *
     * To test enum parameters
     */
    @Test
    public void testEnumParametersTest() {
        List<String> enumFormStringArray = null;
        String enumFormString = null;
        List<String> enumHeaderStringArray = null;
        String enumHeaderString = null;
        List<String> enumQueryStringArray = null;
        String enumQueryString = null;
        Integer enumQueryInteger = null;
        Double enumQueryDouble = null;
        // Void response = api.testEnumParameters(enumFormStringArray, enumFormString, enumHeaderStringArray, enumHeaderString, enumQueryStringArray, enumQueryString, enumQueryInteger, enumQueryDouble);

        // TODO: test validations
    }
    
}

File: samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake_tests/test/api/fake_api_test.dart
import 'dart:typed_data';

import 'package:built_collection/built_collection.dart';
import 'package:dio/dio.dart';
import 'package:http_mock_adapter/http_mock_adapter.dart';
import 'package:openapi/openapi.dart';
import 'package:test/test.dart';

void main() {
  late Openapi client;
  late DioAdapter tester;

  setUp(() {
    client = Openapi(dio: Dio());
    tester = DioAdapter(dio: client.dio);
  });

  tearDown(() {
    tester.close();
  });

  group(FakeApi, () {
    group('testEndpointParameters', () {
      test('complete', () async {
        tester.onPost(
          '/fake',
          (server) => server.reply(200, null),
          data: {
            'number': '3',
            'double': '-13.57',
            'pattern_without_delimiter': 'patternWithoutDelimiter',
            'byte': '0',
            'float': '1.23',
            'integer': '45',
            'int32': '2147483647',
            'int64': '9223372036854775807',
            'date': '2020-08-11',
            'dateTime': '2020-08-11T12:30:55.123Z',
            'binary': '[0, 1, 2, 3, 4, 5]',
          },
          headers: <String, dynamic>{
            'content-type': 'application/x-www-form-urlencoded',
            'content-length': Matchers.integer,
          },
        );

        final response = await client.getFakeApi().testEndpointParameters(
              number: 3,
              double_: -13.57,
              patternWithoutDelimiter: 'patternWithoutDelimiter',
              byte: '0',
              float: 1.23,
              integer: 45,
              int32: 2147483647,
              int64: 9223372036854775807,
              date: Date(2020, 8, 11),
              dateTime: DateTime.utc(2020, 8, 11, 12, 30, 55, 123),
              binary: Uint8List.fromList([0, 1, 2, 3, 4, 5]),
            );

        expect(response.statusCode, 200);
      });

      test('minimal', () async {
        tester.onPost(
          '/fake',
          (server) => server.reply(200, null),
          data: {
            'byte': '0',
            'double': '-13.57',
            'number': '3',
            'pattern_without_delimiter': 'patternWithoutDelimiter',
          },
          headers: <String, dynamic>{
            'content-type': 'application/x-www-form-urlencoded',
            'content-length': Matchers.integer,
          },
        );

        final response = await client.getFakeApi().testEndpointParameters(
              number: 3,
              double_: -13.57,
              patternWithoutDelimiter: 'patternWithoutDelimiter',
              byte: '0',
            );

        expect(response.statusCode, 200);
      });
    });

    group('testEnumParameters', () {
      test('in body data', () async {
        // Not sure if this is correct, we are not sending
        // form data in the body but some weird map
        tester.onGet(
          '/fake',
          (server) => server.reply(200, null),
          data: {
            'enum_form_string': 'formString',
            'enum_form_string_array': Matchers.listParam<Object?>(
              ListParam(
                ['foo', 'bar'],
                ListFormat.csv,
              ),
            ),
          },
          queryParameters: <String, dynamic>{
            'enum_query_string': '-efg',
          },
          headers: <String, dynamic>{
            'enum_header_string': '-efg',
            'content-type': 'application/x-www-form-urlencoded',
            'content-length': Matchers.integer,
          },
        );

        final response = await client.getFakeApi().testEnumParameters(
              enumFormString: 'formString',
              enumFormStringArray: ListBuilder<String>(
                <String>['foo', 'bar'],
              ).build(),
            );

        expect(response.statusCode, 200);
      });

      test('in query parameters', () async {
        tester.onGet(
          '/fake',
          (server) => server.reply(200, null),
          queryParameters: <String, dynamic>{
            'enum_query_string_array': Matchers.listParam<dynamic>(
              ListParam<dynamic>(
                <String>['a', 'b', 'c'],
                ListFormat.multi,
              ),
            ),
            'enum_query_model_array': Matchers.listParam<dynamic>(
              ListParam<dynamic>(
                <String>['_abc', '-efg'],
                ListFormat.multi,
              ),
            ),
            'enum_query_string': 'foo',
            'enum_query_double': 1.23,
            'enum_query_integer': 42,
          },
          headers: <String, dynamic>{
            'enum_header_string': '-efg',
            'content-type': 'application/x-www-form-urlencoded',
          },
          data: <String, dynamic>{},
        );

        final response = await client.getFakeApi().testEnumParameters(
              enumQueryStringArray: ListBuilder<String>(
                <String>['a', 'b', 'c'],
              ).build(),
              enumQueryModelArray: ListBuilder<ModelEnumClass>(
                <ModelEnumClass>[ModelEnumClass.abc, ModelEnumClass.efg],
              ).build(),
              enumQueryString: 'foo',
              enumQueryDouble: 1.23,
              enumQueryInteger: 42,
            );

        expect(response.statusCode, 200);
      });

      test('in header parameters', () async {
        tester.onGet(
          '/fake',
          (server) => server.reply(200, null),
          headers: <String, dynamic>{
            'enum_header_string': 'foo',
            'enum_header_string_array': '[a, b, c]',
            'content-type': 'application/x-www-form-urlencoded',
          },
          data: <String, dynamic>{},
        );

        final response = await client.getFakeApi().testEnumParameters(
              enumHeaderStringArray: ListBuilder<String>(
                <String>['a', 'b', 'c'],
              ).build(),
              enumHeaderString: 'foo',
            );

        expect(response.statusCode, 200);
      });
    });
  });
}
